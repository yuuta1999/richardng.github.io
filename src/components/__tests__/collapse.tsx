import React from 'react'
import { keyframes } from 'styled-components'
import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import Collapse from '../collapse'

afterEach(cleanup)


const showDropdown = keyframes`
  0% {
    opacity: 0;
    transform: translate(0, 10px) perspective(200px) rotateX(-2deg);
    transition: visibility 0.25s, opacity 0.25s, transform 0.25s;
  }

  100% {
    transform: translate(0, 0);
    opacity: 1;
  }
`

const hideDropdown = keyframes`
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0;
    transform: translate(0, 10px);
  }
`

test('it works with normall props', () => {

  const { container } = render(<Collapse show={false} collapsed={false}/>)

  expect(container.firstChild).toMatchSnapshot()
})

test('it works with passed props', () => {
  const { container } = render(<Collapse show={true} collapsed={true}/>)

  expect(container.firstChild).toMatchSnapshot()
})
