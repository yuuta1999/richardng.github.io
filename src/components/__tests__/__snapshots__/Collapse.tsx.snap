// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works with normall props 1`] = `
.c0 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: none;
}

@media screen and (max-width:992px) {
  .c0 {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    z-index: 30;
    margin: 0.5rem;
    overflow-y: auto;
    height: auto;
    padding: 1.5rem;
    border: 1px solid var(--Theme-Navbar--Border__onScroll);
    border-radius: 0.5rem;
    box-shadow: 1px 2px 18px var(--Theme-Navbar--BoxShadow__onScroll);
    background: var(--Theme-Navbar--Background);
    -webkit-animation: fuIBnd 300ms ease forwards;
    animation: fuIBnd 300ms ease forwards;
    opacity: 0;
  }

  .c0 .sc-bdVaJa {
    padding: 0;
  }
}

<div
  class="c0"
/>
`;

exports[`it works with passed props 1`] = `
.c0 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (max-width:992px) {
  .c0 {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    z-index: 30;
    margin: 0.5rem;
    overflow-y: auto;
    height: auto;
    padding: 1.5rem;
    border: 1px solid var(--Theme-Navbar--Border__onScroll);
    border-radius: 0.5rem;
    box-shadow: 1px 2px 18px var(--Theme-Navbar--BoxShadow__onScroll);
    background: var(--Theme-Navbar--Background);
    -webkit-animation: klXEug 300ms ease forwards;
    animation: klXEug 300ms ease forwards;
    opacity: 1;
  }

  .c0 .sc-bdVaJa {
    padding: 0;
  }
}

<div
  class="c0"
/>
`;
